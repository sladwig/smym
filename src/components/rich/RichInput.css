.rich-input {
    border: 1px solid black;
    padding: 5px;
}
.rich-input *:focus {
    outline: none;
}