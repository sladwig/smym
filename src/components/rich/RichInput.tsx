import React, { useState, useRef, useEffect } from 'react';
import './RichInput.css';
import { EditorState } from 'prosemirror-state';
import { EditorView } from 'prosemirror-view';
import applyDevTools from 'prosemirror-dev-tools';

import { schema } from './richSchema';
import { plugins } from './richPlugins';
import { nodeViews } from './nodes/nodes';
import { useUpdateState } from '../../hooks/useUpdateState';
import { useEditorState } from '../../hooks/useEditorState';

export const RichInput = () => {
    const editorRef = useRef<HTMLDivElement>(null);
    const [editorView, setEditorView] = useState<EditorView>();

    const { state, dispatchTransaction, stateRef } = useEditorState(
        EditorState.create({ schema, plugins: plugins() }),
    );

    useEffect(() => {
        if (!editorRef.current) return;

        const view = new EditorView(editorRef.current, {
            nodeViews,
            state: stateRef.current,
            dispatchTransaction,
        });
        setEditorView(view);
        applyDevTools(view);
        return () => view.destroy();
    }, [dispatchTransaction, stateRef]);

    useUpdateState(editorView, state);

    return <div className="rich-input" ref={editorRef} />;
};
