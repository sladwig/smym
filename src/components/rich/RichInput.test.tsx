import { render } from '@testing-library/react';
import React from 'react';
import { RichInput } from './RichInput';

describe('RichInput', () => {
    test('renders', () => {
        const { container } = render(<RichInput />);
        expect(container).toMatchSnapshot();
    });
});
