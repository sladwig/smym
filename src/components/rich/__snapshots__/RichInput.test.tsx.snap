// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichInput renders 1`] = `
<div>
  <div
    class="rich-input"
  >
    <div
      class="ProseMirror"
      contenteditable="true"
    >
      <span
        class="node__dom"
      >
        <span
          style="text-decoration: underline;"
        >
          <span
            class="node__content-dom"
          />
        </span>
      </span>
      <br />
    </div>
  </div>
</div>
`;
